test/archethic/mining/smart_contract_validation_test.exs

```diff
@@ -0,0 +1,148 @@
+
+  import Mox
+
+  describe "fetch_contracts/1" do
+    test "returns a list of valid contracts" do
+      code1 = """
+      condition inherit: [
+        uco_transfers: [%{ to: "3265CCD78CD74984FAB3CC6984D30C8C82044EBBAB1A4FFFB683BDB2D8C5BCF9", amount: 1000000000}],
+        content: "hello"
+      ]
+
+      condition transaction: []
+
+      actions triggered_by: transaction do
+        add_uco_transfer to: "3265CCD78CD74984FAB3CC6984D30C8C82044EBBAB1A4FFFB683BDB2D8C5BCF9", amount: 1000000000
+      end
+      """
+
+      code2 = """
+      condition inherit: [
+        content: regex_match?(\"hello\")
+      ]
+
+      condition transaction: []
+
+      actions triggered_by: transaction do
+        set_content "hello"
+      end
+      """
+
+      code3 = """
+      condition inherit: [
+        content: regex_match?("hello"),
+        uco_transfers: %{"addr3" => 1000000000},
+        type: transfer
+      ]
+
+      condition transaction: []
+
+      actions triggered_by: transaction do
+        add_uco_transfer to: "addr3", amount: 1000000000
+        set_content "hello"
+        set_type transfer
+      end
+      """
+
+      MockClient
+      |> stub(
+        :send_message,
+        fn
+          _, %GetTransaction{address: "addr1"}, _ ->
+            {:ok, %Transaction{data: %TransactionData{code: code1}}}
+
+          _, %GetTransaction{address: "addr2"}, _ ->
+            {:ok, %Transaction{data: %TransactionData{code: code2}}}
+
+          _, %GetTransaction{address: "addr3"}, _ ->
+            {:ok, %Transaction{data: %TransactionData{code: code3}}}
+        end
+      )
+
+      node = %Node{
+        ip: "127.0.0.1",
+        port: 1234,
+        first_public_key: :crypto.strong_rand_bytes(32),
+        last_public_key: :crypto.strong_rand_bytes(32),
+        available?: true,
+        authorized?: true,
+        authorization_date: DateTime.utc_now(),
+        geo_patch: "AAA"
+      }
+
+      P2P.add_and_connect_node(node)
+
+      recipients = ["addr1", "addr2", "addr3"]
+      contracts = recipients |> SmartContractValidation.fetch_contracts() |> Enum.to_list()
+      assert length(contracts) == 3
+    end
+  end
+
+  describe "validate_remote_code/2" do
+    test "returns true if all contracts calls are valid" do
+      contract =
+        %Transaction{
+          address: "@SC1",
+          type: :contract,
+          data: %TransactionData{
+            code: """
+            @version 1
+
+            condition inherit: [
+              content: true
+            ]
+            condition transaction: [
+              content: "hi"
+            ]
+            actions triggered_by: transaction do
+              Contract.set_content "hello"
+            end
+            """
+          }
+        }
+        |> Contract.from_transaction!()
+
+      assert SmartContractValidation.valid_contract_calls?([contract], %Transaction{
+               data: %TransactionData{content: "hi"}
```

Comment:
We'll probably have a failure here when #1020 is merged (interpreter will require a validated transaction (with a validationstamp))